### PR TITLE
docs: add bitbucket samples

### PR DESCRIPTION
**Pull request by viceice**
Add some samples for bitbucket.org pipelines

### CODE REVIEW

**Comment by viceice** [comment]
@HonkingGoose Can you please have some review 🙃

File: docs/bitbucket.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
+# Samples for bitbucket pipelines
```
Comment (HonkingGoose) [inline docs/bitbucket.md:1]:
```suggestion
# Samples for Bitbucket pipelines
```

File: docs/bitbucket.md [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+This are samples for running a self-hosted renovate on bitbucket.org pipelines.
```
Comment (HonkingGoose) [inline docs/bitbucket.md:3]:
```suggestion
These are configuration examples for running a self-hosted Renovate on bitbucket.org pipelines.
```

File: docs/bitbucket.md [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+The branches
```
Comment (HonkingGoose) [inline docs/bitbucket.md:4]: > The branches

I think this needs more text?

File: docs/bitbucket.md [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+            - docker
+          caches:
+            - docker
+          script:
+            - export LOG_LEVEL=debug RENOVATE_CONFIG_FILE="$BITBUCKET_CLONE_DIR/config.js"
+            - renovate
+```
+
+### config.js
+Manual repository config
```
Comment (HonkingGoose) [inline docs/bitbucket.md:39]:
```suggestion
Example of manual repository config:
```

File: docs/bitbucket.md [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+  platform: 'bitbucket',
+  username: process.env.USERNAME,
+  password: process.env.PASSWORD,
+  baseDir: `${process.env.BITBUCKET_CLONE_DIR}/renovate`,
+  autodiscover: true,
+}
+```
+
+### renovate.json
+Use this for self-update renovate
```
Comment (HonkingGoose) [inline docs/bitbucket.md:62]:
```suggestion
Use this for self-update Renovate:
```

File: docs/bitbucket.md [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+      ],
+      "datasourceTemplate": "docker",
+      "versioningTemplate": "docker"
+    }
+  ]
+}
+```
+
+### Variables
+You need to defined pipeline variables
```
Comment (HonkingGoose) [inline docs/bitbucket.md:81]:
```suggestion
You need to define pipeline variables:
```

File: docs/bitbucket.md [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+- `USERNAME`: Bitbucket.org username
+- `PASSWORD`: Bitbucket.org password
+- `GITHUB_COM_TOKEN`: github token to fetch changelog (optional, highly recommend)
```
Comment (HonkingGoose) [inline docs/bitbucket.md:84]:
```suggestion
- `GITHUB_COM_TOKEN`: GitHub token to fetch changelog (optional, highly recommended)
```

File: docs/bitbucket.md (hunk 3 above)
Comment (viceice) [reply docs/bitbucket.md:4]: 🤦‍♂️ Don't remember what i want to write, so removing it
```suggestion
```

File: README.md [hunk 8]
```diff
@@ -14,7 +14,7 @@ For the `latest` image see [here](https://github.com/renovatebot/docker-renovate
 See [docs](https://docs.renovatebot.com/self-hosting/) for additional information to self-hosting renovate with docker.
```
Comment (HonkingGoose) [inline README.md:14]:
```suggestion
See [docs](https://docs.renovatebot.com/self-hosting/) for additional information to self-hosting Renovate with Docker.
```

File: README.md [hunk 9]
```diff
@@ -11,10 +11,10 @@ For the `latest` image see [here](https://github.com/renovatebot/docker-renovate
 
 ## Usage
 
-See [docs](https://docs.renovatebot.com/self-hosting/) for additional information to self-hosting renovate with docker.
+See [docs](https://docs.renovatebot.com/self-hosting/) for additional information to self-hosting Renovate with Docker.
```
Comment (HonkingGoose) [inline README.md:14]:
```suggestion
Read the [self-hosting docs](https://docs.renovatebot.com/self-hosting/) for more information on how to self-host Renovate with Docker.
```

File: docs/bitbucket.md [hunk 10]
```diff
@@ -0,0 +1,87 @@
...
+module.exports = {
+  platform: 'bitbucket',
+  username: process.env.USERNAME,
+  password: process.env.PASSWORD,
+  baseDir: `${process.env.BITBUCKET_CLONE_DIR}/renovate`,
+  repositories: [ "user1/repo1", "orh/repo2" ],
+}
+```
+
+Using autodiscover
```
Comment (HonkingGoose) [inline docs/bitbucket.md:49]:
```suggestion
Using autodiscover:
```

**Review by rarkins — APPROVED** [review]

**Pull request merged**